src/Diagnostics/FxCop/Core/AbstractCodeBlockAnalyzer.cs

```diff
@@ -0,0 +1,22 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.FxCopAnalyzers
+{
+    public abstract class AbstractCodeBlockAnalyzer : DiagnosticAnalyzer
+    {
+        public sealed override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Descriptor);
```

Comment:
Do we really need these Abstract types (this and the SyntaxNode one)? I don't think they add much value - they reduce a couple of lines of typing but I think they just add complexity to the analyzer in that it's all not in one place. Seeing the Initialize method, in my opinion, is a much clearer way of understanding what the analyzer is doing.